pkg/builder/entity_dto_builder.go

```diff
@@ -262,6 +262,20 @@ func (eb *EntityDTOBuilder) ApplicationData(appData *proto.EntityDTO_Application
 	return eb
 }
 
+func (eb *EntityDTOBuilder) VirtualMachineData(vmData *proto.EntityDTO_VirtualMachineData) *EntityDTOBuilder {
+	if eb.err != nil {
+		return eb
+	}
+	if eb.entityDataHasSet {
```

Comment:
Ideally, it should be possible to add properties incrementally to the VirtualMachineData. They may not be available in the single class or method. For this you will need to create a specific VMEntityBuilder (sub classed from EntityDTOBuilder) with methods to add different properties. Do this as a separate restructure task.